aten/src/ATen/native/Sorting.cpp

```diff
@@ -424,13 +462,18 @@ Tensor& quantile_out(
     double q,
     optional<int64_t> _dim,
     bool keepdim) {
+  TORCH_WARN_ONCE(
+      "torch.quantile(input, q, dim=None, keepdim=False, *, out=None) is deprecated "
+      "in favor of the new signature torch.quantile(input, q, dim=None, interpolation='linear', "
+      "keepdim=False, *, out=None) with interpolation parameter.");
   TORCH_CHECK(
       q >= 0 && q <= 1, "quantile() q must be in the range [0, 1] but got ", q);
   return at::quantile_out(
       out,
       self,
       at::scalar_tensor(q, self.options()),
       std::move(_dim),
+      /*interpolation=*/"linear",
       keepdim);
```

Comment:
The specification of the ignore_nan parameter is inconsistent across these calls - seems like it should either always be there or always be absent?

Reply:
ignore_nan parameter only exists in the private quantile_impl function. It's not in the public API.